expression/builtin_time.go

```diff
@@ -5084,21 +5084,22 @@ func isDuration(str string) bool {
 }
 
 // strDatetimeAddDuration adds duration to datetime string, returns a string value.
-func strDatetimeAddDuration(sc *stmtctx.StatementContext, d string, arg1 types.Duration) (string, error) {
+func strDatetimeAddDuration(sc *stmtctx.StatementContext, d string, arg1 types.Duration) (string, bool, error) {
 	arg0, err := types.ParseTime(sc, d, mysql.TypeDatetime, types.MaxFsp)
 	if err != nil {
-		return "", err
+		sc.AppendWarning(err)
```

Comment:
Why don't we check the sql_mode to decide whether to return an error or a warning?

e.g. sc.HandleTruncate

Reply:
Even in the strict mode, MySQL returns a warning.

Reply:
Add a comment for it?